README.md

```diff
@@ -33,161 +33,125 @@ Prerequisites
-   to build a size optimised newlib.
-   ```
-   # Download newlib 3.3 from https://sourceware.org/newlib/
-   $ wget ftp://sourceware.org/pub/newlib/newlib-3.3.0.tar.gz
-   $ tar -xvf newlib-3.3.0.tar.gz
-   $ cd newlib-3.3.0
-   # Disable stdlib for building
-   $ export CFLAGS=-nostdlib
-   # Run configure
-   $ ./configure --disable-newlib-supplied-syscalls --with-gnu-ld --with-newlib --enable-languages=c --target=riscv64-unknown-elf --host=x86 --disable-multi-lib --prefix /usr
-   # Build and then install
-   $ make -j8
-   $ sudo make install
-   ```
+        If your distro doesn't provide a RISC-V toolchain you can build one yourself
+        or just disable RISC-V support by running
 
-   Alternatively, you may use a pre-compiled toolchain that we created with
-   Crosstool-NG.
-   ```
-   $ wget http://cs.virginia.edu/~bjc8c/archive/gcc-riscv64-unknown-elf-8.3.0-ubuntu.zip
-   $ unzip gcc-riscv64-unknown-elf-8.3.0-ubuntu.zip
-   # add gcc-riscv64-unknown-elf-8.3.0-ubuntu/bin to your `$PATH` variable.
-   ```
+        ```shell
+        NORISCV=1 make
+        ```
 
-   **Arch**:
-   ```
-   $ sudo pacman -Syu riscv64-elf-gcc riscv32-elf-newlib arm-none-eabi-newlib riscv64-elf-newlib
-   ```
+    1. libc for embedded targets
 
-   **Fedora**:
+        **Ubuntu (21.10 or later) or Debian (11 or later)**:
 
-   **dnf** does not contain the `riscv-gnu-toolchain`, an alternative is to
-   compile from source. Start with some of the tools we need to compile the
-   source.
-   ```
-   $ sudo dnf install make automake gcc gcc-c++ kernel-devel  texinfo expat expat-devel
-   $ sudo dnf group install "Development Tools" "C Development Tools and Libraries"
-   ```
-   Get `riscv-gnu-toolchain`,  [summarised instructions as stated
-   here](https://github.com/riscv-collab/riscv-gnu-toolchain/blob/master/README.md)
-   ```
-   $ git clone https://github.com/riscv/riscv-gnu-toolchain
-   $ cd riscv-gnu-toolchain/
-   ```
-   **Note: add /opt/riscv/bin to your PATH**, then,
-   ```
-   $ ./configure --prefix=/opt/riscv --enable-multilib
-   ```
-   `--enable-multilib` ensures that "the multilib compiler will have the prefix
-   riscv64-unknown-elf- or riscv64-unknown-linux-gnu- but will be able to target
-   both 32-bit and 64-bit systems."
-   ```
-   $ sudo make         [might need elevated privileges writing to `/opt/`]
-   ```
-   additionally, with
-   ```
-   $ sudo make linux
-   ```
-   you can also build `riscv64-unknown-linux-gnu`, which can be useful with tock
-   where `riscv64-unknown-linux-gnu-objcopy` is used.
+        Use picolibc instead of newlib
 
-   After the the source has been compiled and copied to `/opt/riscv` and
-   `/opt/riscv/bin`has appended to the PATH, the toolchain is ready to be used.
+        ```shell
+        $ sudo apt install picolibc-arm-unknown-elf libnewlib-arm-none-eabi
+        $ sudo apt install picolibc-riscv64-unknown-elf
+        ```
```

Comment:
I'm not clear on what this instruction means. Are users supposed to use picolib?

Also, we compile our own newlib so we can enable PIC. I assume we would need to do that with picolib?

Reply:
I removed the ARM change